include/ftl/task_scheduler.h

```diff
@@ -62,8 +63,8 @@ struct TaskSchedulerInitOptions {
 	unsigned ThreadPoolSize = 0;
 	/* The behavior of the threads after they have no work to do */
 	EmptyQueueBehavior Behavior = EmptyQueueBehavior::Spin;
-	/* Callback to run on thread start */
-	void (*ThreadStartCallback)() = nullptr;
+	/* Callbacks to run at various points to allow for e.g. hooking a profiler to fiber states */
+	ISchedulerUserCallbacks* Callbacks = nullptr;
```

Comment:
If we do end up going the POD struct path from above, then this can be a concrete instance, with just the default constructor. Then no more dereferencing needed